src/test/java/org/example/test.java

```diff
@@ -0,0 +1,74 @@
+package org.example;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.Test;
+
+public class test {
+
+    private final int n = 10;
+    private final double a = -1;
+    private final double b = 1;
+    private final double delta = 0.01;
+
+    private final double[] x = new double[n + 1];
+    private final double[] weights = new double[n + 1];
+
+    private void generateChebyshevPointsAndWeights() {
+        Interpolation.generateChebyshevPoints(n, a, b, x, weights);
+    }
+
+    private double[] evaluateFunction(FunctionEvaluator evaluator) {
+        double[] y = new double[n + 1];
+
+        for (int i = 0; i <= n; i++) {
+            y[i] = evaluator.eval(x[i]);
+        }
+
+        return y;
+    }
+
+    @FunctionalInterface
+    interface FunctionEvaluator {
+        double eval(double x);
+    }
+
+    public test() {
+        generateChebyshevPointsAndWeights();
+    }
+
+    @Test
+    public void testExponentialFunction() {
+        FunctionEvaluator expEvaluator = (xi) -> Math.exp(-xi * xi);
+        double[] y = evaluateFunction(expEvaluator);
+
+        double point = 0.5;
+        double expectedResult = expEvaluator.eval(point);
```

Comment:
zakaj preprosto ne zapišete definicije funkcije?
```java
double expectedResult = Math.exp(-xi * xi);
```

Preveč abstrakcije in indirektnih izračunov škodi, če niso zares nujno potrebni.